File: demo/mobile/todo-kt/app/src/main/java/io/yaver/fixture/native_android/TodoStore.kt
package io.yaver.fixture.native_android

class TodoStore {
  private var nextId = 4

  private val _items = mutableListOf(
    TodoItem(1, "Review remote runtime"),
    TodoItem(2, "Test tap controls"),
    TodoItem(3, "Trigger feedback overlay", done = true),
  )

  val items: List<TodoItem>
    get() = _items.toList()

  fun add(title: String): Boolean {
    val trimmed = title.trim()
    if (trimmed.isEmpty()) return false
    _items.add(0, TodoItem(nextId++, trimmed))
    return true
  }

  fun toggle(id: Int) {
    val index = _items.indexOfFirst { it.id == id }
    if (index >= 0) {
      val item = _items[index]
      _items[index] = item.copy(done = !item.done)
    }
  }

  fun remove(id: Int) {
    _items.removeAll { it.id == id }
  }
}

File: demo/mobile/todo-swift/YaverFixture/TodoStore.swift
import Foundation

@MainActor
final class TodoStore: ObservableObject {
    @Published var draft: String = ""
    @Published private(set) var items: [TodoItem] = [
        TodoItem(title: "Review remote runtime"),
        TodoItem(title: "Test tap controls"),
        TodoItem(title: "Trigger feedback overlay", isDone: true),
    ]

    func addDraft() {
        let trimmed = draft.trimmingCharacters(in: .whitespacesAndNewlines)
        guard !trimmed.isEmpty else { return }
        items.insert(TodoItem(title: trimmed), at: 0)
        draft = ""
    }

    func toggle(_ item: TodoItem) {
        guard let index = items.firstIndex(where: { $0.id == item.id }) else { return }
        items[index].isDone.toggle()
    }

    func remove(_ item: TodoItem) {
        items.removeAll { $0.id == item.id }
    }
}
